src/lib.rs

```diff
@@ -82,22 +77,36 @@ impl RustSemsimian {
         subject_terms: &HashSet<TermID>,
         object_terms: &HashSet<TermID>,
         predicates: &Option<HashSet<Predicate>>,
-    ) -> HashMap<TermID, HashMap<TermID, (f64, f64)>> {
+    ) -> HashMap<TermID, HashMap<TermID, (f64, f64, f64, HashSet<TermID>)>> {
         let self_shared = Arc::new(Mutex::new(self.clone()));
```

Comment:
I was playing around [here](https://github.com/monarch-initiative/semsimian/pull/62) with RwLock instead of Arc/Mutex for this, in case we don't get a massive speedup using this solution (no action required, just while I'm thinking of it)